tests/Integration/UI/MetadataRendererTest.php

```diff
@@ -0,0 +1,229 @@
+	 * @uses \Parsely\Parsely::api_key_is_set
+	 * @uses \Parsely\Parsely::convert_jsonld_to_parsely_type
+	 * @uses \Parsely\Parsely::get_options
+	 * @uses \Parsely\Parsely::post_has_trackable_status
+	 * @uses \Parsely\UI\Metadata_Renderer::filter_empty_and_not_string_from_array
+	 */
+	public function test_render_metadata_repeated_metas(): void {
+		self::set_options( array( 'apikey' => 'testkey' ) );
+
+		$post_id = self::factory()->post->create();
+		$this->go_to( home_url( '/?p=' . $post_id ) );
+
+		ob_start();
+		self::$metadata_renderer->render_metadata( 'repeated_metas' );
+		$out = ob_get_clean();
+
+		self::assertStringContainsString( '<meta name="parsely-type" content="post" />', $out );
+		self::assertStringContainsString( '<meta name="parsely-section" content="Uncategorized" />', $out );
+	}
+
+	/**
+	 * Check that the renderer completes successfully and outputs repeated metas when the current
+	 * post is an integer but not WP_Post object.
+	 * We're not fully testing the output, since that's not the renderer's concern.
+	 *
+	 * @since 3.4.0
+	 *
+	 * @covers \Parsely\UI\Metadata_Renderer::render_metadata
+	 * @uses \Parsely\UI\Metadata_Renderer::__construct
+	 * @uses \Parsely\Metadata::__construct
+	 * @uses \Parsely\Metadata::construct_metadata
+	 * @uses \Parsely\Metadata\Metadata_Builder::__construct
+	 * @uses \Parsely\Metadata\Metadata_Builder::build_basic
+	 * @uses \Parsely\Metadata\Metadata_Builder::clean_value
+	 * @uses \Parsely\Metadata\Metadata_Builder::get_current_url
+	 * @uses \Parsely\Metadata\Post_Builder::__construct
+	 * @uses \Parsely\Metadata\Post_Builder::build_article_section
+	 * @uses \Parsely\Metadata\Post_Builder::build_author
+	 * @uses \Parsely\Metadata\Post_Builder::build_headline
+	 * @uses \Parsely\Metadata\Post_Builder::build_image
+	 * @uses \Parsely\Metadata\Post_Builder::build_keywords
+	 * @uses \Parsely\Metadata\Post_Builder::build_main_entity
+	 * @uses \Parsely\Metadata\Post_Builder::build_metadata_post_times
+	 * @uses \Parsely\Metadata\Post_Builder::build_publisher
+	 * @uses \Parsely\Metadata\Post_Builder::build_thumbnail_url
+	 * @uses \Parsely\Metadata\Post_Builder::build_type
+	 * @uses \Parsely\Metadata\Post_Builder::build_url
+	 * @uses \Parsely\Metadata\Post_Builder::get_author_names
+	 * @uses \Parsely\Metadata\Post_Builder::get_bottom_level_term
+	 * @uses \Parsely\Metadata\Post_Builder::get_category_name
+	 * @uses \Parsely\Metadata\Post_Builder::get_coauthor_names
+	 * @uses \Parsely\Metadata\Post_Builder::get_metadata
+	 * @uses \Parsely\Metadata\Post_Builder::get_tags
+	 * @uses \Parsely\Parsely::api_key_is_missing
+	 * @uses \Parsely\Parsely::api_key_is_set
+	 * @uses \Parsely\Parsely::convert_jsonld_to_parsely_type
+	 * @uses \Parsely\Parsely::get_options
+	 * @uses \Parsely\Parsely::post_has_trackable_status
+	 * @uses \Parsely\UI\Metadata_Renderer::filter_empty_and_not_string_from_array
+	 */
+	public function test_render_metadata_global_post(): void {
```

Comment:
Maybe a name indicating that we're testing with an integer post would be suitable here?